src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs

```diff
@@ -14,119 +16,447 @@ namespace Microsoft.SqlTools.ServiceLayer.Connection.Contracts
-        public string AttachDbFilename { get; set; }
+        public string AttachDbFilename
+        {
+            get
+            {
+                return getOptionValue<string>("attachDbFilename");
+            }
+
+            set
+            {
+                setOptionValue("attachDbFilename", value);
+            }
+        }
 
         /// <summary>
         /// Gets or sets the name or address of the partner server to connect to if the primary server is down.
         /// </summary>
-        public string FailoverPartner { get; set; }
+        public string FailoverPartner
+        {
+            get
+            {
+                return getOptionValue<string>("failoverPartner");
+            }
+
+            set
+            {
+                setOptionValue("failoverPartner", value);
+            }
+        }
 
         /// <summary>
         /// If your application is connecting to an AlwaysOn availability group (AG) on different subnets, setting MultiSubnetFailover=true provides faster detection of and connection to the (currently) active server.
         /// </summary>
-        public bool? MultiSubnetFailover { get; set; }
+        public bool? MultiSubnetFailover
+        {
+            get
+            {
+                return getOptionValue<bool?>("multiSubnetFailover");
+            }
+
+            set
+            {
+                setOptionValue("multiSubnetFailover", value);
+            }
+        }
 
         /// <summary>
         /// When true, an application can maintain multiple active result sets (MARS).
         /// </summary>
-        public bool? MultipleActiveResultSets { get; set; }
+        public bool? MultipleActiveResultSets
+        {
+            get
+            {
+                return getOptionValue<bool?>("multipleActiveResultSets");
+            }
+
+            set
+            {
+                setOptionValue("multipleActiveResultSets", value);
+            }
+        }
 
         /// <summary>
         /// Gets or sets the size in bytes of the network packets used to communicate with an instance of SQL Server.
         /// </summary>
-        public int? PacketSize { get; set; }
+        public int? PacketSize
+        {
+            get
+            {
+                return getOptionValue<int?>("packetSize");
+            }
+
+            set
+            {
+                setOptionValue("packetSize", value);
+            }
+        }
 
         /// <summary>
         /// Gets or sets a string value that indicates the type system the application expects.
         /// </summary>
-        public string TypeSystemVersion { get; set; }
+        public string TypeSystemVersion
+        {
+            get
+            {
+                return getOptionValue<string>("typeSystemVersion");
+            }
+
+            set
+            {
+                setOptionValue("typeSystemVersion", value);
+            }
+        }
+
+        private T getOptionValue<T>(string name)
```

Comment:
if the method is called with nullable type then the default is null. For example GetOptionValue<int?>('some param') returns null if the option is not set